### ISSUE TITLE
Port scan does not show all open ports

### ISSUE THREAD

**Issue by jsveiga** [aaronjwood/PortAuthority#19]
Hello, I'm using Port Authority 1.5.0 (thank you) on a Nexus, 5 5.1.1.

When I run a port scan on a known host, not all open ports are shown.

Even if I use the Scan Port Range specifically covering the open ports, they do not show on the list, although I can see in the host (tshark) that it did SYN,ACK the probes.

For example, Scan Well Known Ports only gives me:
143 - imap
22 - ssh
25 - smtp
53 - domain
80 - http

Scan Port Range from 1 - 1024 results in the same list.

Nevertheless, I have more ports open as the nmap (from other host in the same subnet) shows:
PORT     STATE SERVICE
22/tcp   open  ssh
25/tcp   open  smtp
53/tcp   open  domain
80/tcp   open  http
143/tcp  open  imap
443/tcp  open  https
587/tcp  open  submission
993/tcp  open  imaps
3128/tcp open  squid-http
3306/tcp open  mysql
5222/tcp open  xmpp-client
9090/tcp open  zeus-admin
9091/tcp open  xmltec-xmlmail

If I look at tshark on the target host, while doing for example a Port Authority scan in the range 9089-9092, I get this:

Capturing on 'eth1'
  1   0.000000 192.168.168.30 -> 192.168.168.1 TCP 74 46612→9089 [SYN] Seq=0 Win=65535 Len=0 MSS=1460 SACK_PERM=1 TSval=13716725 TSecr=0 WS=64
  2   0.000044 192.168.168.1 -> 192.168.168.30 TCP 54 9089→46612 [RST, ACK] Seq=1 Ack=1 Win=0 Len=0
  3   0.000171 192.168.168.30 -> 192.168.168.1 TCP 74 60554→9091 [SYN] Seq=0 Win=65535 Len=0 MSS=1460 SACK_PERM=1 TSval=13716725 TSecr=0 WS=64
  4   0.000216 192.168.168.1 -> 192.168.168.30 TCP 74 9091→60554 [SYN, ACK] Seq=0 Ack=1 Win=28960 Len=0 MSS=1460 SACK_PERM=1 TSval=670147978 TSecr=13716725 WS=128
  5   0.000556 192.168.168.30 -> 192.168.168.1 TCP 74 55836→9090 [SYN] Seq=0 Win=65535 Len=0 MSS=1460 SACK_PERM=1 TSval=13716725 TSecr=0 WS=64
  6   0.000589 192.168.168.1 -> 192.168.168.30 TCP 74 9090→55836 [SYN, ACK] Seq=0 Ack=1 Win=28960 Len=0 MSS=1460 SACK_PERM=1 TSval=670147978 TSecr=13716725 WS=128
  7   0.000611 192.168.168.30 -> 192.168.168.1 TCP 74 35060→9092 [SYN] Seq=0 Win=65535 Len=0 MSS=1460 SACK_PERM=1 TSval=13716725 TSecr=0 WS=64
  8   0.000638 192.168.168.1 -> 192.168.168.30 TCP 54 9092→35060 [RST, ACK] Seq=1 Ack=1 Win=0 Len=0
  9   0.002448 192.168.168.30 -> 192.168.168.1 TCP 66 60554→9091 [ACK] Seq=1 Ack=1 Win=87616 Len=0 TSval=13716726 TSecr=670147978
 10   0.002565 192.168.168.30 -> 192.168.168.1 TCP 66 55836→9090 [ACK] Seq=1 Ack=1 Win=87616 Len=0 TSval=13716726 TSecr=670147978
 11   0.002807 192.168.168.30 -> 192.168.168.1 TCP 66 60554→9091 [FIN, ACK] Seq=1 Ack=1 Win=87616 Len=0 TSval=13716726 TSecr=670147978
 12   0.002937 192.168.168.30 -> 192.168.168.1 TCP 66 55836→9090 [FIN, ACK] Seq=1 Ack=1 Win=87616 Len=0 TSval=13716726 TSecr=670147978
 13   0.003144 192.168.168.1 -> 192.168.168.30 TCP 66 9090→55836 [FIN, ACK] Seq=1 Ack=2 Win=29056 Len=0 TSval=670147979 TSecr=13716726
 14   0.005853 192.168.168.1 -> 192.168.168.30 TCP 66 9091→60554 [FIN, ACK] Seq=1 Ack=2 Win=29056 Len=0 TSval=670147979 TSecr=13716726
 15   0.006044 192.168.168.30 -> 192.168.168.1 TCP 66 55836→9090 [ACK] Seq=2 Ack=2 Win=87616 Len=0 TSval=13716726 TSecr=670147979
 16   0.007130 192.168.168.30 -> 192.168.168.1 TCP 66 60554→9091 [ACK] Seq=2 Ack=2 Win=87616 Len=0 TSval=13716726 TSecr=670147979

Server responded on ports 9090 and 9091, but they do not show on Port Authority as opened. 

I can also access the server from the phone at port 9091 (Openfire administration web interface), so there is no blocking between them.

If you need any extra debugging info please let me know.

Best regards,
Joao S Veiga

**Comment by aaronjwood** [comment]
Thanks for reporting this and sending all of that info over. I'll take a look into this.

**Comment by aaronjwood** [comment]
This was caused by a really stupid issue when reading the port data CSV file that I took from IANA (http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml). Not only is there spotty data but their CSV formatting is extremely inconsistent in some areas! I was not handling errors around this as well as I should have which is why the application looked like it was skipping the step of adding the open port to the list. This should be fixed now. See this https://github.com/aaronjwood/PortAuthority/commit/a4dc506abe39f360ceb8a14dd933927fd7e80bc0

Going to push out a new version with this fix right now. Thanks again for reporting this!
